Document/0x04g-Testing-Cryptography.md

```diff
@@ -142,9 +142,9 @@ Please note that the usage of IVs is different when using CTR and GCM mode in wh
 
 #### Padding Oracle Attacks due to Weaker Padding or Block Operation Implementations
 
-In the old days, PKCS #7 (Public Key Cryptography Standards 7) was used as a padding mechanism when doing asymmetric encryption. Now in modern Java environments it is referred to as PKCS #5. This mechanism is vulnerable to the padding oracle attack. Therefore, it is best to use OEAP (Optimal Asymmetric Encryption Padding) (or PKCS #1 v2.0). Note that, even when using OAEP, you can still run into an issue known best as the Mangers attack as described [in the blog at Kudelskisecurity](https://research.kudelskisecurity.com/2018/04/05/breaking-rsa-oaep-with-mangers-attack/ "Kudelskisecurity").
+In the old days, PKCS1.5 padding (`PKCS1Padding`) was used as a padding mechanism when doing asymmetric encryption. This mechanism is vulnerable to the padding oracle attack. Therefore, it is best to use OEAP (Optimal Asymmetric Encryption Padding) (or PKCS #1 v2.0) (possible alternatives: `OAEPPadding`, `OAEPwithSHA-256andMGF1Padding`,	`OAEPwithSHA-224andMGF1Padding`, `OAEPwithSHA-384andMGF1Padding`, `OAEPwithSHA-512andMGF1Padding`). Note that, even when using OAEP, you can still run into an issue known best as the Mangers attack as described [in the blog at Kudelskisecurity](https://research.kudelskisecurity.com/2018/04/05/breaking-rsa-oaep-with-mangers-attack/ "Kudelskisecurity").
```

Comment:
```suggestion
In the old days, PKCS1.5 padding (`PKCS1Padding`) was used as a padding mechanism when doing asymmetric encryption. This mechanism is vulnerable to the padding oracle attack. Therefore, it is best to use OAEP (Optimal Asymmetric Encryption Padding) (or PKCS #1 v2.0) (possible alternatives: `OAEPPadding`, `OAEPwithSHA-256andMGF1Padding`,	`OAEPwithSHA-224andMGF1Padding`, `OAEPwithSHA-384andMGF1Padding`, `OAEPwithSHA-512andMGF1Padding`). Note that, even when using OAEP, you can still run into an issue known best as the Mangers attack as described [in the blog at Kudelskisecurity](https://research.kudelskisecurity.com/2018/04/05/breaking-rsa-oaep-with-mangers-attack/ "Kudelskisecurity").
```

There are 3x elements in parenthesis one after another.

- Are OAEP or PKCS the alternatives? Or is it another name for the same thing? Is yes. I'd write aka.
- what's marked as alternatives: is there another name for.them? Modes, transformations, variants? For example, you could put a separate sentence instead of the parenthesis. "The OAEP variants include: `OAEP...`

Reply:
rephrased it a bit :)